internal/handlers/human_resource_handlers.go

```diff
@@ -384,6 +395,61 @@ func (h *Handler) PatchHumanResource(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+	// Determine if this PATCH only updates the allowed trio of fields:
+	// status, is_completed, headcount_got. If so, we'll bypass PIN verification.
+	onlyAllowedTrio := func(in humanResourcePatchInput) bool {
+		// Track if any of the allowed fields are actually being updated
+		hasAnyAllowed := false
+		if in.Status != nil {
+			hasAnyAllowed = true
+		}
+		if in.IsCompleted != nil {
+			hasAnyAllowed = true
+		}
+		if in.HeadcountGot != nil {
+			hasAnyAllowed = true
+		}
+
+		// If any other updatable field is present in the payload, it's not a limited update
+		if in.Org != nil || in.Address != nil || in.Phone != nil || in.HasMedical != nil || in.PiiDate != nil ||
+			in.RoleName != nil || in.RoleType != nil || in.Skills != nil || in.Certifications != nil ||
+			in.ExperienceLevel != nil || in.LanguageRequirements != nil || in.HeadcountNeed != nil ||
+			in.HeadcountUnit != nil || in.RoleStatus != nil || in.ShiftStartTs != nil || in.ShiftEndTs != nil ||
+			in.ShiftNotes != nil || in.AssignmentTimestamp != nil || in.AssignmentCount != nil ||
+			in.AssignmentNotes != nil || in.TotalRolesInRequest != nil || in.CompletedRolesInRequest != nil ||
+			in.PendingRolesInRequest != nil || in.TotalRequests != nil || in.ActiveRequests != nil ||
+			in.CompletedRequests != nil || in.CancelledRequests != nil || in.TotalRoles != nil ||
+			in.CompletedRoles != nil || in.PendingRoles != nil || in.UrgentRequests != nil ||
+			in.MedicalRequests != nil {
+			return false
+		}
+		return hasAnyAllowed
+	}(in)
+	// Fetch stored pin (if any)
+	var storedPin *string
+	if err := h.pool.QueryRow(context.Background(), `select valid_pin from human_resources where id=$1`, id).Scan(&storedPin); err != nil {
+		if err == pgx.ErrNoRows {
+			c.JSON(http.StatusNotFound, gin.H{"error": "not found"})
+			return
+		}
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	// PIN behavior:
+	// - If VERIFY_HR_PIN=false, do not enforce matching; still generate/set one if missing to onboard records.
+	// - If VERIFY_HR_PIN=true, enforce match once a pin exists; else generate/set as needed.
+	if storedPin == nil || strings.TrimSpace(*storedPin) == "" {
+		// bypass
+	} else {
+		// PIN exists already
+		if os.Getenv("VERIFY_HR_PIN") == "true" && !onlyAllowedTrio {
```

Comment:
we could check this condition before loading the pin code from the DB to reduce the number of queries.